### PR TITLE
Cso

### PR DESCRIPTION
**Pull request by Laura-Jane**
cats are moving
finished from my point of view

### CODE REVIEW

File: CatSwarmOptimization.ts.old [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import * as mersenneTwister from './node_modules/mersenne-twister';
```
Comment (raspyweather) [inline CatSwarmOptimization.ts.old]: File should not exist

File: src/ClassicalSeekingMode.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        const copies: Cat[] = this.createCopies(cat);
+        for (let i = 0; i < this.j; i++) {
+            this.changePosition(copies[i]);
+            this.fitnessValues[i] = copies[i].calculateFitness();
+        }
+        if (this.selfPositionConsidering) {
+            this.fitnessValues[this.j] = copies[this.j].calculateFitness();
+        }
+        for (let i = 1; i < this.fitnessValues.length; i++) {
+            allTheSame = (this.fitnessValues[i - 1] === this.fitnessValues[i]);
```
Comment (raspyweather) [inline src/ClassicalSeekingMode.ts]: variable allTheSame really necessary?

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                break;
+            } else {
+                if (bestPosition === undefined) {
+                    console.log('no bestposition!');
+                }
+                tracingMode.trace(cats[i], bestPosition);
+            }
+        }
+        lastTwentyCats[iterationCounter % 20] = fitnessValueBest;
+        if (iterationCounter > 19) {
```
Comment (raspyweather) [inline src/index.ts]: why 19? what is this magic number doing?

File: src/ClassicalSeekingMode.ts (hunk 2 above)
Comment (Laura-Jane) [reply src/ClassicalSeekingMode.ts]: put check into the method, so the variable doesn't need to be tranferred

File: src/index.ts (hunk 3 above)
Comment (Laura-Jane) [reply src/index.ts]: put this absolute number into parameters.json and config, so it is a parameter now

**Pull request merged**